airflow/contrib/hooks/gcp_dataflow_hook.py

```diff
@@ -34,8 +34,12 @@
 
 # This is the default location
 # https://cloud.google.com/dataflow/pipelines/specifying-exec-params
+
 DEFAULT_DATAFLOW_LOCATION = 'us-central1'
 
+# These are the supported python interpreters
+DATAFLOW_SUPPORTED_PY_INTERPRETERS = {'python2', 'python3'}
```

Comment:
I am not familiar with airflow, but wondering if we should also include `python`, `python3.5`, `python3.6`, `python3.7` if users wish to have less or more control as to which interpreter to choose.

Reply:
@tvalentyn Good call out on minor versions. 

Currently [only python 3.5 is considered stable for beam.](https://beam.apache.org/roadmap/python-sdk/#python-3-support)
However, adding minor version to interpreter might be good.

This gets a little tricky because we don't know what is available in whatever system is running airflow (could be on a standalone VM, on Kubernetes, in a Hadoop cluster, etc. etc.). 

In [cloud composer it will be 2.7.15 or 3.6.6](https://cloud.google.com/composer/docs/concepts/python-version).

By not hard coding minor version it softens assumptions about the infra that composer is running on.

Not sure what the best path forward is here.

Reply:
Thanks for pointing out Beam webpage - it is a little bit out of date, and I'll fix it. The upcoming release 2.14.0 will officially announce support for 3.6, 3.7, however the Python 3 differences between 2.13.0 and 2.14.0 are minor, and we should not restrict to Airflow to Python 3.5 because of a dependency on Beam.

Reply:
Is there a way we can submit a python dataflow job w/o having to invoke the pipeline script client side? 
If we could just provide a payload to Dataflow API saying "hey use this python version and run this python file (ie. on gcs)". 
This might allow for a more flexible design for this operator (though a significant re structuring).

Reply:
How frequently do we change the pipeline? If infrequent, you can consider creating a template: https://cloud.google.com/dataflow/docs/guides/templates/executing-templates#example-1-custom-template-batch-job. 
Template would tied to Python minor version it was created with.

Reply:
I will add the minor versions.
Re: templates there's a separate dataflow operator for templates. (Which I should also update as part of this pr as well thanks for pointing that out) will push these changes tomorrow.